chain/chain/src/chain.rs

```diff
@@ -807,6 +807,8 @@ impl Chain {
         (
             ShardChunk,
```

Comment:
I think it's time to create a `struct` for this. It becomes confusing what is what here.

Reply:
++